### ISSUE TITLE
Add a log-logistic distribution fitting

### ISSUE THREAD

**Issue by monocongo** [monocongo/climate_indices#106]
[It's been shown](http://spei.csic.es/home.html#p9) that a log-logistic distribution fitting is optimal for SPEI calculations. Let's add this to complement the gamma and Pearson III distribution fittings we already have available. This may already be available in some form in [statsmodels ](https://www.statsmodels.org/stable/index.html)or [SciPy](https://www.scipy.org/), we should investigate there before rolling our own.

**Comment by monocongo** [comment]
Apparently, the log-logistic distribution is also referred to as the [Fisk distribution](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.fisk.html#scipy.stats.fisk), available within SciPy.

**Comment by monocongo** [comment]
Related to issue #190.

**Comment by Emmadd** [comment]
I created a draft (there are some minor issues) pull request for a Fisk implementation in #440
However, looking into the R code some more, I realised that the R SPEI package uses a generalised logistic distribution, more similar to [genlogistic](https://docs.scipy.org/doc/scipy/reference/reference/generated/scipy.stats.genlogistic.html#scipy.stats.genlogistic) in SciPy than to Fisk. The distributions might be similar when c=-/k, but I am not certain.

**Comment by monocongo** [comment]
Thanks for your work on this @Emmadd 

Is it useful for us to also provide a generalized logistic distribution? I'd like to include it if so, but only if truly useful as it will add maintenance cost, and this project is already due significant maintenance, we don't want to add even more technical debt.

**Comment by Emmadd** [comment]
I don't think so. But if someone with more statistical knowledge thinks otherwise it would be good to know.

**Comment by Emmadd** [comment]
After some further investigation it seems the python SciPy Fisk option is the better one to use. The genlogistic in python is a special generalised logistic distribution, not THE generalised logistic distribution that R uses, which resembles Fisk. So #440 can be continued.

**Comment by monocongo** [comment]
## Feasibility assessment (2026-09-22, against `main` b079dee7)

Short version: the plumbing is small now; the hard part is deciding **which** log-logistic, and picking wrong locks a public enum name onto the wrong distribution. Discussion opened at #1126 — please weigh in, especially if you have used these distributions in practice.

### The trap

This thread (and closed PR #440) assumes SciPy `fisk` is the distribution R `SPEI` fits. It is not:

- R `SPEI` 1.8.1's `"log-Logistic"` branch calls `pelglo`/`parglo` + `lmom::cdfglo` — Hosking's **generalized logistic (GLO)** — with `ub-pwm` L-moments by default. SPEIbase v2.11 is built with that code.
- SciPy `fisk` is the 3-parameter log-logistic, and `fisk.fit` is MLE only; xclim standardizes with that. `scipy.stats.genlogistic` is a third family again.
- Illustrative check, 60 values per calendar month of P−PET-like data, GLO L-moments vs `fisk.fit`: mean/max `|ΔSPEI|` ≈ 0.06/0.25 (wet), 0.16/1.0 (dry), 0.23/1.24 (skewed). For a drought index the tails are the product.

### What is cheap now

`main` has since unified the SPI/SPEI fit seam in `compute.fit_and_standardize`, so a third distribution is:

- one dispatch branch in that seam;
- `loglogistic_parameters` + `transform_fitted_loglogistic` in `compute.py`;
- an L-moment estimator in `lmoments.py` (`_estimate_lmoments` already exists; GLO parameters are ~8 lines from `lmom`'s `PELGLO`);
- `Distribution` member, validation text, fitting-parameter key set, spatial parameter prep.

The CLI and xarray metadata pick up new `Distribution` members automatically. Roughly 250–350 LOC plus tests, 1–2 sessions. PR #440 should not be resurrected as-is: it predates the seam and its `scipy.stats.fisk.fit` estimator is the fragile, slow part. Open PR #1124 currently documents log-logistic as unavailable, so coordinate if that merges first.

### Needed before implementation

1. GLO vs Fisk vs both, and L-moments vs MLE ([discussion #1126](https://github.com/monocongo/climate_indices/discussions/1126)).
2. Public enum name: `loglogistic`, `fisk`, or `glo`.
3. SPI as well as SPEI, or SPEI-only? SPI precipitation zeros need a probability mass that the SPEI offset P−PET does not.
4. Independent validation oracle: an R `SPEI` install, or the committed SPEIbase fixtures (currently compared against gamma with #106 noted as the confound).
